Character.h

```diff
@@ -1,17 +1,23 @@
+#ifndef CHARACTER_H
+#define CHARACTER_H
+
 #include <string>
 #include <fstream>
 
 class Character
 {
-private:
+protected:
 	const std::string name;
-	unsigned int HP;
-	const unsigned int DMG;
+	float maxHP, HP, DMG;
+	const float getDMG() const;
+	static std::string* parseUnit(const std::string &fileName);
+	Character(const std::string* characterAttributes);
 public:
-	Character(const std::string characterName, const unsigned int healthPoints, const unsigned int damagePoints);
-	static Character parseUnit(std::string fileName);
+	static Character CharacterFromFile(const std::string &fileName);
 	const std::string getName() const;
-	const unsigned int getHP() const;
-	const unsigned int getDMG() const;
-	void gotHit(const Character &attacker);
+	const float getHP() const;
+	float gotHit(Character* attacker);
+	virtual void doHit(Character& victim);

```

Comment:
Nem jó, hogy mindkét metódus public, így nem egyértelmű az osztály használójának, hogy melyiket (esetleg mindkettőt) kéne használnia.

Reply:
Igen, ezt elfelejtettem áttenni. Javítva.

Reply:
Oh, már emlékszem miért volt public. Hiszen a victimre hívja meg a doHit(). Azért van ez a gotHit() - doHit() páros, mert az a gotHit a sebzettet módosítja, a doHit() a támadót módosít(hat)ja. Hogyan lehet protected a gotHit(), de mégis megoldva a sebzés a doHit()-en belül?

Reply:
Egyébként az a veszély, hogy valaki mindkettőt használná, tényleg fennáll. A másik commentben említett egymástól független sebző és XP kapó függvények megoldanák ezt a problémát, csak szerintem sok más szempontból bonyolultabb lenne a szerkezet. Például az is könnyen okozna akkor félreértést, hogy miért van gainXP() normál karakterre, akinek még csak nincs is XP-je. (Persze mert a fight alatt nem tudjuk, hogy ki Player és ki nem.)
   
Másfelől a külön használható gotHit() kimondottan hasznosnak tűnik.   
    
Nem jó úgy, ha írok kommenteket a Character.h-ba? Vagy később nem terjed ki úgyis erre is a Doxygen?

Reply:
>Nem jó úgy, ha írok kommenteket a Character.h-ba? Vagy később nem terjed ki úgyis erre is a Doxygen?

A Doxygen nem erre való. Olyan metódusnak nem szabadna publikusnak lennie, amit nem szabad meghívni kívülről.

Viszont amire gondoltam, az valóban nem működik itt, mert különböző osztályokról van szó. (Nem volt előírva, hogy újat kell származtatni, elég lett volna kiegészíteni a Charactert.) Mivel úgyis át kell adni egy támadót a `gotHit`nek, így nem olyan vészes, maradhat. Másik lehetőség a setter (`setHP`) használata lenne, de akkor meg azt lehetne bármivel meghívni. Elfogadom így.

Reply:
Igazából pont ma akartam megkérdezni a társam, hogy mit csinál a Doxygen, amin dolgozik. Akkor helyette hozzáadok pár sor kommentet a headerbe.